modules/tinydnsbackend/tinydnsbackend.cc

```diff
@@ -171,17 +171,25 @@ void TinyDNSBackend::getAllDomains(vector<DomainInfo>* domains, bool include_dis
 
   while (get(rr)) {
     if (rr.qtype.getCode() == QType::SOA && dupcheck.insert(rr.qname).second) {
-      SOAData sd;
-      fillSOAData(rr.content, sd);
-
       DomainInfo di;
       di.id = -1; //TODO: Check if this is ok.
       di.backend = this;
       di.zone = rr.qname;
-      di.serial = sd.serial;
-      di.notified_serial = sd.serial;
       di.kind = DomainInfo::Master;
       di.last_check = time(0);
+
+      if (getSerial) {
+        SOAData sd;
+        try {
+          fillSOAData(rr.content, sd);
+          di.serial = sd.serial;
+        }
+        catch (PDNSException) {
```

Comment:
CodeQL reported:  This should catch a PDNSException by (const) reference rather than by value. 

Also, if we can be specific (PDNSException) here, why catch `...` in bindbackend?

Reply:
The (existing) code in bind backend is completely different. It is catching exceptions in the local getSOA() function vs fillSOAData() here.